Jenkinsfile

```diff
@@ -14,28 +14,54 @@ pipeline {
   parameters {
     choice(name: 'deployProfile',
       description: 'Choose where the built plugin should be deployed to',
-      choices: ['sonatype.snapshots', 'maven.central.release'])
-
-    string(name: 'revision',
-      description: 'Revision for this release, e.g. newest version "1.3.0" revision should be "1" (-> 1.3.1). Note: This is only used for release target!',
-      defaultValue: '0' )
+      choices: ['central.snapshots', 'maven.central.release'])
   }
 
   stages {
     stage('build') {
       steps {
         script {
+          def targetBranch = "${env.BRANCH_NAME}-${env.BUILD_NUMBER}"
+          def officialRelease = params.deployProfile == 'maven.central.release';
+          
+          if (officialRelease) {
+            sh "git config --global user.name 'ivy-team'"
+            sh "git config --global user.email 'info@ivyteam.ch'"
+            
+            sh "git checkout -b ${targetBranch}"
+            sh "git tag -l | xargs git tag -d"
+            
+            def releasedVersion = version('-DremoveSnapshot=true')
+            sh "git add . ;git commit -m 'update versions for release ${releasedVersion}'"
+            sh "git tag \"v${releasedVersion}\" -a -m \"Official release ${releasedVersion}\""
+          }
```

Comment:
just what maven-release-plugin does; without all the often failing magic :magic_wand: